centreon/www/front_src/src/Header/Resources/Host/getHostPropsAdapter.ts

```diff
@@ -0,0 +1,212 @@
+import numeral from 'numeral';
+
+import { SeverityCode } from '@centreon/ui';
+import type { SelectEntry } from '@centreon/ui';
+
+import {
+  getHostResourcesUrl,
+  upCriterias,
+  unreachableCriterias,
+  downCriterias,
+  pendingCriterias,
+  unhandledStateCriterias,
+  hostCriterias
+} from '../getResourcesUrl';
+import getDefaultCriterias from '../../../Resources/Filter/Criterias/default';
+import type { Adapter } from '../useResourceCounters';
+import type { Criteria } from '../../../Resources/Filter/Criterias/models';
+import type { SubMenuProps } from '../../sharedUI/ResourceSubMenu';
+import type { CounterProps } from '../../sharedUI/ResourceCounters';
+import type { HostStatusResponse } from '../../api/decoders';
+
+import {
+  labelDownStatusHosts,
+  labelUnreachableStatusHosts,
+  labelUpStatusHosts,
+  labelAll,
+  labelDown,
+  labelPending,
+  labelUnreachable,
+  labelUp
+} from './translatedLabels';
+
+type ChangeFilterAndNavigate = (params: {
+  criterias: Array<Criteria>;
+  link: string;
+}) => (e: React.MouseEvent<HTMLLinkElement>) => void;
+
+export interface HostPropsAdapterOutput {
+  counters: CounterProps['counters'];
+  hasPending: boolean;
+  items: SubMenuProps['items'];
+}
+
+type GetHostPropsAdapter = Adapter<HostStatusResponse, HostPropsAdapterOutput>;
+
+const formatCount = (number: number | string): string => numeral(number).format('0a')
```

Comment:
As you declare the two functions for Host and Service, it might ba a good idea to extract it into a utils files in order to avoid to duplicate this function

Reply:
@Thebarda là je pense que vu le diff avant/après du code des tests etc, on ferait bien de prendre le win, car je m'approche dangereusement du désespoir à ce stade. 

Ça pourrait aussi être pas mal de mettre quelques règles de base concernant les CR, à mon avis il faut faire quelques passes dessus, genre 2max pour une petite PR, 3 pour une grosse. 

à discuter. 

Certaines refactos n'autorisent pas les découpage de PR/tickets à volonté, mais si on met trop de temps à les merger ou qu'on se montre trop perfectionnistes, on risque de désespérer tout développeur à prendre des risques pour dé-endetter la code base. 

Pour moi le plus important ICI, c'est d'avoir une bonne couverture de test, une bonne SOC, KISS et iso design.

Je ne dit pas que le retour n'est pas cohérent, mais je dois être à ma 15° correction poussée et les retours arrivent encore.